File: native/src/main/scala/io/github/edadma/suit/InterFont.scala
package io.github.edadma.suit

import scala.scalanative.unsafe.*

// The InterVariable font, embedded directly in the binary. Scala Native has no runtime
// resource loading — a native executable has no classpath — so a bundled asset is linked in
// as C data at compile time (the bytes in resources/scala-native/inter_data.c, reached
// through the accessors in inter_font.c) and read from memory here. This is what lets Inter
// be suit's default font with no external file dependency, identical on every platform.
// Inter is a variable font with a `wght` axis; the bytes are loaded into one face per weight
// with FreeType's newMemoryFace (see [[Fonts]]). Inter is under the SIL Open Font License
// (fonts/OFL.txt).
@extern
object InterFont:
  def suit_inter_font_data(): Ptr[Byte] = extern
  def suit_inter_font_size(): CLong     = extern

File: native/src/main/resources/scala-native/inter_font.c
// Accessors for the embedded InterVariable font (the bytes live in inter_data.c). Scala
// Native has no runtime resource loading — a native executable has no classpath — so the
// font is linked in as C data at compile time and read from memory. Exposing it through
// two small functions keeps the Scala extern (see InterFont.scala) free of array-symbol
// linkage details. The font is loaded into a face with FreeType newMemoryFace (see Fonts);
// InterVariable is under the SIL Open Font License (fonts/OFL.txt).
extern unsigned char suit_inter_data[];
extern unsigned int  suit_inter_data_len;

const unsigned char *suit_inter_font_data(void) { return suit_inter_data; }
long suit_inter_font_size(void) { return (long)suit_inter_data_len; }
